h2o-algos/src/main/java/hex/tree/DTree.java

```diff
@@ -1035,20 +1081,59 @@ static Split findBestSplitPoint(DHistogram hs, int col, double min_rows,
                   // Or tied SE, then pick split towards middle bins
                   (selo + sehi == best_seL + best_seR &&
                           Math.abs(b - (nbins >> 1)) < Math.abs(best - (nbins >> 1)))) {
-            if( (wlo[b] + wNA) >= min_rows && whi[b] >= min_rows) {
-              if (constraint == 0 || (constraint * (nomlo[b] + nomNA) / (denlo[b] + denNA) <= constraint * nomhi[b] / denhi[b])) {
+            if((wlo[b] + wNA) >= min_rows && whi[b] >= min_rows) {
+              double tmpPredLeft;
+              double tmpPredRight;
+              if(constraint != 0 && dist._family.equals(DistributionFamily.quantile)) {
+                  int quantileBinLeft = 0;
+                  int quantileBinRight = 0;
+                  double ratio = 1;
+                  double delta = 1;
+                  for (int bin = 1; bin <= nbins; bin++) {
+                    // left tree prediction quantile
+                    if (bin <= b) {
+                      if (bin == 1) {
+                        ratio = 100 / wlo[b];
+                      }
+                      double tmpQuantile = (wlo[bin] * ratio) / 100;
+                      assert tmpQuantile <= 1 + 10e-5 : "Calculated temporary quantile " + tmpQuantile + " cannot be higher than 1.";
+                      double tmpDelta = Math.abs(dist._quantileAlpha - tmpQuantile);
+                      if (tmpDelta < delta) {
+                        delta = tmpDelta;
+                        quantileBinLeft = bin;
+                      }
+                      // right tree prediction quantile
+                    } else {
+                      if (bin == b + 1) {
+                        delta = 1;
+                        ratio = 100 / (wlo[nbins] - wlo[b]);
+                      }
+                      double tmpQuantile = ((wlo[bin] - wlo[b]) * ratio) / 100;
+                      assert tmpQuantile <= 1 + 10e-5 : "Calculated temporary quantile " + tmpQuantile + " cannot be higher than 1.";
```

Comment:
10e-5 == 1e-04, was that intentional?

good thinking on using precision

Reply:
Fixed.